civis/tests/test_futures.py

```diff
@@ -239,6 +261,100 @@ def test_polling_interval(self, *mocks):
 
         clear_lru_cache()
 
+    @mock.patch(api_import_str, return_value=civis_api_spec_base)
+    @mock.patch.object(CivisFuture, '_subscribe')
+    def test_as_completed(self, *mocks):
+        clear_lru_cache()
+        my_futures = [QUEUED_FUTURE, CANCELLED_FUTURE, FINISHED_FUTURE]
+        fs = futures.as_completed(my_futures)
+        f1 = next(fs)
+        f2 = next(fs)
+        finished_futures = set([f1, f2])
+
+        self.assertEqual(finished_futures,
+                         set([FINISHED_FUTURE, CANCELLED_FUTURE]))
+        clear_lru_cache()
+
+    @mock.patch(api_import_str, return_value=civis_api_spec_base)
+    @mock.patch.object(CivisFuture, '_subscribe')
+    def test_wait(self, *mocks):
+        clear_lru_cache()
+        done, not_done = futures.wait([QUEUED_FUTURE, FINISHED_FUTURE],
+                                      return_when=futures.FIRST_COMPLETED)
+        self.assertEqual(set([FINISHED_FUTURE]), done)
+        self.assertEqual(set([QUEUED_FUTURE]), not_done)
+        clear_lru_cache()
+
+    @mock.patch(api_import_str, return_value=civis_api_spec_base)
+    @mock.patch.object(CivisFuture, '_subscribe')
```

Comment:
Same as my comments for `create_civis_future`, I think that this test will be better if you use `create_client_mock` and avoid patching.